tools/super-tox.py

```diff
@@ -252,40 +269,41 @@ async def super_tox(
         queue.put_nowait((repo, environment, results))
 
     tasks = []
-    for i in range(workers):
-        task = asyncio.create_task(
-            worker(f"worker-{i}", queue, conf, ops_source, ops_source_branch, mode)
-        )
+    for i in range(settings.workers):
+        task = asyncio.create_task(worker(f"worker-{i}", queue, conf))
         tasks.append(task)
     await queue.join()
     for task in tasks:
         task.cancel()
+
     await asyncio.gather(*tasks)
+    results.sort(key=lambda d: d["location"])
     success_count = 0
+
     for result in results:
         try:
             if not result["passed"]:
                 continue
+        # FIXME these exceptions never happen...
         except (TypeError, asyncio.CancelledError) as e:
             logger.error("Task was cancelled: %s", e, exc_info=True)
             continue
         except Exception as e:
             logger.error("Unable to process result: %s", e, exc_info=True)
             continue
         success_count += 1
-    print(
-        f"{success_count} out of {len(results)} ({success_count / len(results)}%) runs passed."
-    )
-
-
-async def worker(
-    name,
-    queue,
-    conf,
-    ops_source: str,
-    ops_source_branch: str | None,
-    mode: typing.Literal["local"] | typing.Literal["lxd"],
-):
+    pct = 100 * success_count // len(results)
+    print(f"{success_count} out of {len(results)} ({pct}%) runs passed.")
+    if settings.verbose:
+        print("Failed for these repos:")
+        pprint.pprint([
+            str(d["location"].relative_to(settings.cache_folder))
+            for d in results
+            if not d["passed"]
+        ])
```

Comment:
I think this is fine for now, but in general I'd rather avoid `--verbose` and `--debug` type flags in favour of logging levels and using the logging module rather than `print/pprint` (I realise I'm breaking my own rule with the print just above this).

I've generally been using a log level of warning to get a list of the failures. This works fine too. What I'd like is for it to be much smarter and give the failure details somewhere (in the actual output or in the log seems too nosy, so maybe in some sort of supporting files folder?) but I haven't had time to do that.

Reply:
My reason to do this was to be able to seed `super-tox.toml` effectively:
- run with blank config against mainline ops
- chuck failing repos into the config
- run with new config against some ops PR or branch

WDYT?

Reply:
Feels like it would be simple enough to get that from logged output too, and maybe there should actually be some sort of "initialise the toml" command that would do this for you.

But fine with this for now.